follow/models.py

```diff
@@ -0,0 +1,160 @@
+        try:
+            relation = Follow.objects.get(follower=follower, followee=followee)
+            try:
+                true_friend_bidirect = Follow.objects.get(follower=followee, followee=follower)
+                if relation.true_friend and true_friend_bidirect.true_friend:
+                    true_friend_bidirect.true_friend = True
+                    true_friend_bidirect.save()
+            except Follow.DoesNotExist:
+                pass
+            relation.delete()
+            return True
+        except Follow.DoesNotExist:
+            return False
+
+    def check_follow(self, follower, followee):
+        try:
+            relation = Follow.objects.get(follower=follower, followee=followee)
+            return True
+        except Follow.DoesNotExist:
+            return False
+
+    def check_true_friend(self, follower, followee):
+        if Follow.objects.filter(
+            follower=follower,
+            followee=followee,
+            true_friend=True).exists() and Follow.objects.filter(
+                follower=followee,
+                followee=follower,
+                true_friend=True).exists():
+            return True
+        else:
+            return False
+
+
+class Follow(models.Model):
+    followee = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="followee")
+    follower = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="follower")
+    created = models.DateTimeField(default=timezone.now)
+    true_friend = models.BooleanField(default=False)
+    objects = FollowManager()
+
+    class Meta:
+        unique_together = ("followee", "follower")
+
+    def __str__(self):
+        return f"{self.followee.Username} is followed by {self.follower.Username}"
+
+    def save(self, *args, **kwargs):
+        if self.followee == self.follower:
+            raise ValidationError("User cannot follow themselves.")
+        super().save(*args, **kwargs)
+
+
+class Request(models.Model):
+    from_user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="from_user")
+    to_user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="to_user")
+    created = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        unique_together = ("from_user", "to_user")
+
+    def __str__(self):
+        return f"{self.from_user.Username}'s request to follow {self.to_user.Username}"
+
+    def accept(self):
+        relation, created = Follow.objects.get_or_create(follower=self.from_user, followee=self.to_user)
+
+        if created is False:
+            raise AlreadyExistsError(f"{self.from_user.Username} has already followed {self.to_user.Username}")
+
+        request_accept.send(sender=self, from_user=self.from_user, to_user=self.to_user)
+
+        try:
+            relation_reverse = Follow.objects.get(follower=self.to_user, followee=self.from_user)
+            relation_reverse.real_friend = True
```

Comment:
Where is `real_friend` defined?

Reply:
This was supposed to be `true_friend` but from our discussion, we'll just query if a user is a true friend on demand using `Follow.object.check_true_friend(user1, user2)`